provider/cmd/pulumi-gen-kubernetes/main.go

```diff
@@ -63,69 +63,90 @@ const (
 	DotNet Language = "dotnet"
 	Go     Language = "go"
 	NodeJS Language = "nodejs"
+	Kinds  Language = "kinds"
 	Python Language = "python"
 	Schema Language = "schema"
 )
 
 func main() {
 	if len(os.Args) < 4 {
-		log.Fatal("Usage: gen <language> <swagger-file> <root-pulumi-kubernetes-dir>")
+		log.Fatal("Usage: gen <language> <swagger-or-schema-file> <root-pulumi-kubernetes-dir>")
 	}
 
 	language := Language(os.Args[1])
 
-	swagger, err := ioutil.ReadFile(os.Args[2])
-	if err != nil {
-		panic(err)
-	}
-
-	swaggerDir := filepath.Dir(os.Args[2])
-
-	legacySwaggerPath := filepath.Join(swaggerDir, Swagger117FileName)
-	err = DownloadFile(legacySwaggerPath, Swagger117Url)
-	if err != nil {
-		panic(err)
-	}
-	legacySwagger, err := ioutil.ReadFile(legacySwaggerPath)
-	if err != nil {
-		panic(err)
-	}
-	mergedSwagger := mergeSwaggerSpecs(legacySwagger, swagger)
-	data := mergedSwagger.(map[string]interface{})
-
 	BaseDir = os.Args[3]
 	TemplateDir = path.Join(BaseDir, "provider", "pkg", "gen")
 	outdir := path.Join(BaseDir, "sdk", string(language))
 
-	// Generate schema
-	pkgSpec := gen.PulumiSchema(data)
-
-	// Generate package from schema
-	pkg := genPulumiSchemaPackage(pkgSpec)
-
-	// Generate provider code
-	genK8sResourceTypes(pkg)
-
 	switch language {
 	case NodeJS:
 		templateDir := path.Join(TemplateDir, "nodejs-templates")
-		writeNodeJSClient(pkg, outdir, templateDir)
+		writeNodeJSClient(readSchema(os.Args[2]), outdir, templateDir)
```

Comment:
Rather than using `os.Args[2]`, can you assign to a variable so it's a little clearer what's getting passed in here?

Reply:
Sure.